Initial commit of all files for todoist noteplan sync

dbludeau opened the pull request.

jgclark commented on dbludeau.TodoistNoteplanSync/CHANGELOG.md:
Even if it just says "0.1.0" and "First release" or something, this needs to be updated before first release.

jgclark commented on dbludeau.TodoistNoteplanSync/plugin.json:
```diff
@@ -0,0 +1,162 @@
+  "plugin.id": "dbludeau.TodoistNoteplanSync",
+  "plugin.name": "🧩 Todoist Noteplan Sync",
+  "plugin.version": "0.1.0",
+  "plugin.lastUpdateInfo": "Describe this update",
```
"First release"

jgclark commented on dbludeau.TodoistNoteplanSync/requiredFiles/html-plugin-comms.js:
Looks like you can remove this file.

jgclark commented on dbludeau.TodoistNoteplanSync/src/NPMessagesFromHTMLWindow.js:
Looks like you can remove this file.

jgclark reviewed (commented):
Well done for all this work, and these updates, which now use the right structure to suit the release process that we have built up over time.
There are a couple of small things mentioned here. 
But there is one significant question for you: the main code, pushed in your first PR, doesn't have flow typing information in it. I'm guessing that's just because you're not familiar with it. All the other plugins use it* and so it would be better for you to adopt it early on. It was new to me whenI started to build plugins, and sometimes its annoying, but overall it really helps spot bugs before compile and test.

* apart from a few bits of code that run in an HTML Window which can't

dbludeau commented:
I think I stumbled my way around git to get the updated version pushed for review.  You are correct, I am not familiar with Flow (or that much javascript).  I got it to where the Flow console is not complaining any more, but not sure if that gets everything you want to see fixed.  Let me know what else needs to be done to get the initial release out there.

jgclark commented on dbludeau.TodoistNoteplanSync/src/NPPluginMain.js:
I see you've made code style changes. This is different from flow, which is about adding types. So, very roughly, function definitions (both parameters are returns) and variable initialisation need to have a ": <type>" added after them. (See any plugin for examples.)
You also want to have JSDoc in the comment block before each function. This is just good coding practice to help future you and other potential future maintainers. (See any plugin for examples.)
Also, I think you'll find that all `logError()` and similar calls need to have an extra parameter. Parameter 2 is what you've been using, but you need parameter 1 to be the imported `pluginJson` object, or a text string, perhaps "TodoistSync0.1".

dbludeau replied:
Alright.  I get there eventually (then know what I am supposed to do for future pushes).  I did add a couple of return types, but oddly flow complained about them saying that needed TypeScript, so I pulled them out.  I'll get those in there.  I understand that concept.  I'll fix the log calls and add the JSDoc comments as well. Thanks for your patience on this.

jgclark replied:
No problem -- there's a lot to get used to. 
I've turned off TS support in my code editor (VSCode) and I never get any comments about TS. Are you running flow in the editor or from the command line?

dbludeau commented:
I’m running it in vscode following your setup guide. I’ll play with the settings. It’s not a big deal as the code still compiles fine, but confused me a little at first. On Aug 19, 2023, at 12:18 PM, Jonathan Clark ***@***.***> wrote:﻿
@jgclark commented on this pull request.



On dbludeau.TodoistNoteplanSync/src/NPPluginMain.js:
No problem -- there's a lot to get used to.
I've turned off TS support in my code editor (VSCode) and I never get any comments about TS. Are you running flow in the editor or from the command line?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

dwertheimer commented:
@dbludeau FWIW you can copy/paste each function to chatGPT and ask it to write Flow types and JSDoc. It may not understand the intent of the function, but it will get it 80% right. Will save you a lot of time. And studying the answers will help you learn Flow. :)

dbludeau commented:
Lets try one more time.  If you see some more issues for clean up, a specific example in my code and a specific example of correct code would be a big help.  Flow is very confusing. (@jgclark  I had to laugh.  I had the FLOW coverage toggled off last time, so no wonder I didn't even really know it was complaining)

dbludeau commented:
If this has not been reviewed, I can push up a 0.2 version (or try to figure out how to merge that branch with this one).  It has a couple more commands and a little bit more code organization and reuse.  Its basically ready to go after I pop a couple more JSDocs in there.  Let me know if want to just deal with this and I can push that as a separate pull request or if you want me to combine.

dwertheimer commented:
@dbludeau you should be able to push your changes up to this same branch on github and the changes will be reflected in the PR

dbludeau commented:
@dwertheimer changes are commited (so much easier just using the source control in VSCode then trying to mess with the command line).  Thanks guys!  Hopefully getting closer.

dbludeau commented:
One last change to add an additional command.  I have the ones in there that I want now, so no more pushes from me while I wait to hear back from you guys.

jgclark commented:
I can confirm the plugin builds and installs OK for me, but I can't test further as I don't use Todoist.
But I did spot that most of the command names start "todoist" but two start "Todoist" ...
Let me know when you're ready for me to release it publicly.

dbludeau commented:
thanks @jgclark.  I cleaned up those command names and changed up the documentation a little.  I pushed it to my fork, but not sure what needs to be done to get it here since this pull is closed.  I can do another pull request it that is the way to go.  I am happy to have you release it once we get those changes integrated.

jgclark commented:
Hi. If you've made changes then please make another PR, and I'll commit and release.
Once we've gone through this first public release we'll see about getting you access to the repo to make this much easier next time.

dbludeau commented:
@jgclark sounds good.  I will do that Monday.  I am out of town this weekend, so no reason to push when I can't help if someone has issues trying to use it.
